src/client/dfuse/il/int_read.c

```diff
@@ -15,22 +15,54 @@
 static ssize_t
 read_bulk(char *buff, size_t len, off_t position, struct fd_entry *entry, int *errcode)
 {
-	daos_size_t read_size = 0;
-	d_iov_t     iov       = {};
-	d_sg_list_t sgl       = {};
-	int         rc;
+	daos_size_t	read_size = 0;
+	d_iov_t		iov       = {};
+	d_sg_list_t	sgl       = {};
+	daos_event_t	ev;
+	bool		flag = false;
+	int		rc;
 
 	DFUSE_TRA_DEBUG(entry->fd_dfsoh, "%#zx-%#zx", position, position + len - 1);
 
 	sgl.sg_nr = 1;
 	d_iov_set(&iov, (void *)buff, len);
 	sgl.sg_iovs = &iov;
-	rc = dfs_read(entry->fd_cont->ioc_dfs, entry->fd_dfsoh, &sgl, position, &read_size, NULL);
+
+	if (daos_handle_is_inval(ioil_eqh))
+		daos_eq_create(&ioil_eqh);
```

Comment:
Need to handle errors in case daos_eq_create() fails.

Reply:
not really, because if it fails, we fall back to the original way of using the internal one.

Reply:
We do however want to avoid the case where it's attempted, and fails for every I/O.

I'm not sure why this needs to be called for every read and write however, can't this be done in the fork handler or are you thinking of the case where a program calls fork and exec but you don't want to do anything before the exec?  In which so I think a boolean flag or lock indicates that this call needs to be made was appropriate.  Alternatively, you could do in on open, this might miss the case where files where open during the fork and then written to before another file was opened but that's probably an acceptable cost.

Reply:
Ah, I'd missed that this was on a thread local variable.